src/block-management/block-manager.js

```diff
@@ -159,6 +159,11 @@ export default class BlockManager extends React.Component<PropsType, StateType>
 	mergeBlocks( clientId: string, forward: boolean ) {
 		// find currently focused block
 		const focusedItemIndex = this.getDataSourceIndexFromClientId( clientId );
+		if ( focusedItemIndex === -1 ) {
+			// do nothing if it's not found.
```

Comment:
Let's expand the comment a bit, to include how can this happen (update form native arriving late and block already been deleted). Same with the other comments below in the diff.

We'll need to pay attention to this while we will be moving to the new stores (GB ones) so, a descriptive comment here will help us there. Thanks!

Reply:
I've updated those comments. 👍